include/af/array.h

```diff
@@ -12,10 +12,57 @@
 #include <af/seq.h>
 #include <af/util.h>
 #include <af/index.h>
+#include <af/exception.h>
 
 #ifdef __cplusplus
 #include <af/traits.hpp>
 #include <vector>
+#ifdef AF_ENABLE_STL_CAST
+#include <valarray>
+#endif
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+    /**
+        \ingroup construct_mat
+        @{
+    */
+    /**
+       Create an \ref af_array handle initialized with user defined data
+
+       This function will create an \ref af_array handle from the memory provided in \p data
+
+       \param[out]  arr The pointer to the returned object.
+       \param[in]   data The data which will be loaded into the array
+       \param[in]   ndims The number of dimensions read from the \p dims parameter
+       \param[in]   dims A C pointer with \p ndims elements. Each value represents the size of that dimension
+       \param[in]   type The type of the \ref af_array object
+
+       \returns \ref AF_SUCCESS if the operation was a success
+    */
+    AFAPI af_err af_create_array(af_array *arr, const void * const data, const unsigned ndims, const dim_t * const dims, const af_dtype type);
+    /**
+      @}
+    */
+
+    /**
+       \ingroup method_mat
+       @{
+    */
+    /**
+       \brief Reduce the reference count of the \ref af_array
+    */
+    AFAPI af_err af_release_array(af_array arr);
```

Comment:
They are used in the new assignment operators and need to be defined before since the new operators are templated.